SignalServiceKit/src/Messages/OWSMessageManager.m

```diff
@@ -139,6 +139,13 @@ - (void)yapDatabaseModified:(NSNotification *)notification
 {
     if (AppReadiness.isAppReady) {
         [OWSMessageUtils.sharedManager updateApplicationBadgeCount];
+    } else {
+        static dispatch_once_t onceToken;
```

Comment:
This doesn't fix any known issues, but I realized that anywhere we abort because `isAppReady` isn't true yet, we should enqueue a block that will try again when the app is ready.  Slightly laborious, but that will make this code easier to reason about.  I use a once token to prevent enqueuing the block more than once.